### PR TITLE
U4 222 - password reset functionality

### PR DESCRIPTION
**Pull request by AndyButland**
This pull request provides password reset functionality for back-office users that have forgotten their password.  It uses the methods available via ASP.Net identity, to generate a reset code that is emailed to the user.  They can then click on that, it's validated and if valid they can set a new password.

### CODE REVIEW

**Comment by Rodske** [comment]
Where did this PR get to? @Shazwazza, do you think it's liked to get pull into 7.4.2?

File: src/Umbraco.Web.UI/Controllers/AccountSurfaceController.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Umbraco.Web.UI.Controllers
+{
+    using System.Web.Mvc;
+    using System.Web.Security;
+    using Mvc;
+
+    public class AccountSurfaceController : SurfaceController
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI/Controllers/AccountSurfaceController.cs]: Pretty sure this has this been committed by accident?

**Comment by Shazwazza** [comment]
Hi, no it will not make it into 7.4.2. I've been on holidays and we haven't had time to fully review this yet. It's of course a much needed feature, i'll schedule this for next sprint.

File: src/Umbraco.Web.UI/Controllers/AccountSurfaceController.cs (hunk 1 above)
Comment (AndyButland) [inline src/Umbraco.Web.UI/Controllers/AccountSurfaceController.cs]: Yes, looks like I was testing something out.  Away myself at the moment but if you don't do anything further with this by the end of the week I'll clean this up.

**Comment by kipusoep** [comment]
Awesome work @AndyButland! We've been wanting this for a long long time! :+1:

**Comment by AndyButland** [comment]
@kipusoep - yes, think that's the lowest issue number (222) that I've looked at, was a while ago it was first requested.  It's in an upcoming sprint now I see so hopefully it's OK to be pulled in soon.

**Comment by Shazwazza** [comment]
Thanks @AndyButland  I've manually pulled this in it is now in a 7.5 branch. I'll comment on the original issue with feedback.

**Pull request closed without merging**